packages/react-components/react-aria/src/activedescendant/useOptionWalker.ts

```diff
@@ -0,0 +1,75 @@
+import * as React from 'react';
+import { useFluent_unstable as useFluent } from '@fluentui/react-shared-contexts';
+import { isHTMLElement, useIsomorphicLayoutEffect } from '@fluentui/react-utilities';
+
+interface UseOptionWalkerOptions {
+  matchOption: (el: HTMLElement) => boolean;
+}
+
+export function useOptionWalker<TListboxElement extends HTMLElement>(options: UseOptionWalkerOptions) {
+  const { matchOption } = options;
+  const { targetDocument } = useFluent();
+  const treeWalkerRef = React.useRef<TreeWalker | null>(null);
+  const listboxRef = React.useRef<TListboxElement | null>(null);
+
+  const optionFilter = React.useCallback(
+    (node: Node) => {
+      if (isHTMLElement(node) && matchOption(node)) {
+        return NodeFilter.FILTER_ACCEPT;
+      }
+
+      return NodeFilter.FILTER_SKIP;
+    },
+    [matchOption],
+  );
+
+  useIsomorphicLayoutEffect(() => {
+    if (!targetDocument || !listboxRef.current) {
+      return;
+    }
+
+    treeWalkerRef.current = targetDocument.createTreeWalker(listboxRef.current, NodeFilter.SHOW_ELEMENT, optionFilter);
+  }, [targetDocument, optionFilter]);
+
+  const optionWalker = React.useMemo(
+    () => ({
+      first: () => {
+        if (!treeWalkerRef.current) {
+          return;
```

Comment:
```suggestion
          return null;
```

Let's also return `null`, so we always return `null` or `HTMLElement`